src/fiber/graph.rs

```diff
@@ -837,34 +798,43 @@ where
             .collect::<HashMap<_, _>>();
 
         let mut target = target;
-        let mut current_expiry = final_tlc_expiry_delta;
+        let mut accumulated_expiry = final_tlc_expiry_delta;
+        let accumulated_amount = amount;
         let mut last_edge = None;
 
         if route_to_self {
-            let (new_target, expiry, edge) =
-                self.adjust_target_for_route_self(&hop_hint_map, amount, source, target)?;
-            target = new_target;
-            last_edge = edge;
-            current_expiry += expiry;
+            let (edge, expiry) = self.adjust_target_for_route_self(
+                &hop_hint_map,
+                amount,
+                final_tlc_expiry_delta,
+                source,
+                target,
+            )?;
+            target = edge.source;
```

Comment:
can we add a assertion `assert_ne!(target, edge.source)` to make sure we pick a different target after `adjust_target_for_route_self`.